### PR TITLE
KOGITO-1798: Disable example failing tests with persistence until def…

### PR DESCRIPTION
**Pull request by Sgitario**
…ects are fixed

JIRA Ticket: https://issues.redhat.com/browse/KOGITO-1798
Description: We have to ignore a couple of tests because some defects with persistance.
Also, we wanted to make the quarkus and spring boot versions behave the same using: same name files and using JUnit 5.
Finally, I added some configuration I needed for running Infinispan in Spring Boot (in Quarkus, this configuration was not necessary).

### CODE REVIEW

**Comment by sutaakar** [comment]
@Sgitario IMHO it would be better to introduce a test category disabling the specified tests just when persistence is applied. Otherwise we lose test coverage.

**Comment by Sgitario** [comment]
We cannot use the test category here since the same test case works for either persistence or not persistence.
In order to disable the tests, I made use of the surefire plugin to add a system property and then disable the concrete test methods using the @DisabledIfSystemProperty from JUnit 5.

**Comment by sutaakar** [comment]
one side note, can you please use spaces in pom.xml rather than tabs, to keep content and formatting consistent?

File: process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
+package org.kie.kogito.examples.demo;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Comment (sutaakar) [inline process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java:5]: Can you please unify assetions?
You seems to use old and new JUnit at the same time.

File: process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+		assertEquals(1, orderItemProcesses.values().size());
+
+		ProcessInstance<?> childProcessInstance = orderItemProcesses.values().iterator().next();
+
+		List<WorkItem> workItems = childProcessInstance.workItems(policy);
+		assertEquals(1, workItems.size());
+
+		childProcessInstance.completeWorkItem(workItems.get(0).getId(), null, policy);
+
+		assertEquals(ProcessInstance.STATE_COMPLETED, childProcessInstance.status());
```
Comment (sutaakar) [inline process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java:75]: Why did you delete `assertEquals(ProcessInstance.STATE_COMPLETED, processInstance.status());`

Comment (Sgitario) [reply process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java:75]: It was failing and looking at the Quarkus version, this line was removed as part of https://issues.redhat.com/browse/KOGITO-173.

File: process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java (hunk 1 above)
Comment (Sgitario) [reply process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java:5]: Done

File: process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java (hunk 2 above)
Comment (sutaakar) [reply process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java:75]: hmm, ok
but why it doesn't fail in the test below? The line is there.

File: process-quarkus-example/pom.xml [hunk 3]
```diff
@@ -129,6 +129,19 @@
...
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-surefire-plugin</artifactId>
+            <configuration>
+              <systemPropertyVariables>
+                <tests.category>persistence</tests.category>
+              </systemPropertyVariables>
+            </configuration>
+	      </plugin>
```
Comment (sutaakar) [inline process-quarkus-example/pom.xml:142]:
```suggestion
	    </plugin>
```

File: process-springboot-example/pom.xml [hunk 4]
```diff
@@ -119,6 +119,9 @@
           <name>persistence</name>
         </property>
       </activation>
+      <properties>
+      	<tests.category>persistence</tests.category>
```
Comment (sutaakar) [inline process-springboot-example/pom.xml]:
```suggestion
        <tests.category>persistence</tests.category>
```

File: process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java (hunk 2 above)
Comment (Sgitario) [reply process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java:75]: I didn't take a look, but I'm sure this depends on retriggering processes.

File: process-quarkus-example/pom.xml (hunk 3 above)
Comment (Sgitario) [reply process-quarkus-example/pom.xml:142]: Done

File: process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java (hunk 2 above)
Comment (sutaakar) [reply process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java:75]: ok, can you please take a look then?

File: process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestTest.java [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = DemoApplication.class)
+@DirtiesContext(classMode = ClassMode.BEFORE_EACH_TEST_METHOD) // reset spring context after each test method
```
Comment (sutaakar) [inline process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestTest.java]: Wouldn't it have more sense to keep there AFTER_EACH_TEST_METHOD?

File: process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestTest.java [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+
+	@LocalServerPort
+	int port;
+
+	@BeforeEach
+	public void setUp() {
+		RestAssured.port = port;
+
+        // need it when running with persistence
+        orderProcess.instances().values().forEach(pi -> pi.abort());
```
Comment (sutaakar) [inline process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestTest.java:51]: IMHO it should be rather placed in AfterEach so the test cleans up after it has been executed.

Comment (Sgitario) [reply process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestTest.java:51]: In Quarkus version, it was being done before each test and it does not really matter if done before or after....

File: process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestTest.java (hunk 5 above)
Comment (Sgitario) [reply process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestTest.java]: changed .....

File: process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestTest.java (hunk 6 above)
Comment (sutaakar) [reply process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestTest.java:51]: Yes, though it may affect other tests when the cleanup is done in Before phase.

File: process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java (hunk 2 above)
Comment (Sgitario) [reply process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java:75]: Raised: https://issues.redhat.com/browse/KOGITO-1848

File: process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestTest.java (hunk 6 above)
Comment (Sgitario) [reply process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestTest.java:51]: It's not going to be an issue since no more tests are using the order process, but will do.

**Review by sutaakar — APPROVED** [review]

File: process-springboot-example/src/main/resources/application.properties [hunk 7]
```diff
@@ -2,4 +2,8 @@ spring.mvc.servlet.path=/docs
 
 resteasy.jaxrs.scan-packages=org.kie.kogito.**
 
-infinispan.remote.server-list=127.0.0.1:11222
\ No newline at end of file
+infinispan.remote.server-list=127.0.0.1:11222
```
Comment (radtriste) [inline process-springboot-example/src/main/resources/application.properties]: did you add the auth yourself ?

File: process-springboot-example/pom.xml (hunk 4 above)
Comment (radtriste) [inline process-springboot-example/pom.xml]: any reason for double `tests.category` ?

**Review by radtriste — COMMENTED** [review]
2 small comments below

File: process-springboot-example/src/main/resources/application.properties (hunk 7 above)
Comment (Sgitario) [reply process-springboot-example/src/main/resources/application.properties]: Removed ...

File: process-springboot-example/pom.xml (hunk 4 above)
Comment (Sgitario) [reply process-springboot-example/pom.xml]: fixed

**Review by radtriste — APPROVED** [review]

File: process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestTest.java (hunk 6 above)
Comment (cristianonicolai) [reply process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestTest.java:51]: I usually prefer to have the clean up done in the before method. That helps if you run let's say a separated service ( infinispan ? ) so the state is always the expected one when the test executes.

File: process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java [hunk 8]
```diff
@@ -0,0 +1,129 @@
+package org.kie.kogito.examples.demo;
```
Comment (cristianonicolai) [inline process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java:1]: missing headers?

Comment (Sgitario) [reply process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java:1]: I don't see any header in the process-quarkus-example and process-springboot-example projects at least. Should we add them to all the sources? If so, I guess this should be done as part of another task. What do you think about?

File: process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestTest.java (hunk 6 above)
Comment (Sgitario) [reply process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestTest.java:51]: I agree with you. Usually, I do this stuff in the before methods as a setup configuration of my test suite. What do you think about @sutaakar ? Let me know whether reverting this or keeping it like this.

Comment (sutaakar) [reply process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestTest.java:51]: @Sgitario I don't have a strong preference in that, if you think it is better to do the cleanup before then feel free to do so.

Comment (Sgitario) [reply process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestTest.java:51]: In order to avoid to have an afterEach and a beforeEach both in the Spring Boot examples, I've changed it to be done in the beforeEach method as at the beginning.

File: process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java (hunk 8 above)
Comment (Sgitario) [reply process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersProcessTest.java:1]: Raised a ticket for adding the headers: https://issues.redhat.com/browse/KOGITO-1866

**Review by cristianonicolai — APPROVED** [review]
